screeps-profiler.js

```diff
@@ -191,17 +219,31 @@ const Profiler = {
     { name: 'RoomPosition', val: RoomPosition },
     { name: 'Source', val: Source },
     { name: 'Flag', val: Flag },
+    { name: 'Market', val: Game.market },
   ],
 
-  record(functionName, time) {
+  record(myparent, functionName, time) {
     if (!Memory.profiler.map[functionName]) {
       Memory.profiler.map[functionName] = {
-        time: 0,
+        time: [],
```

Comment:
But a very needed feature otherwise -- current profiler makes it hard to nail down what takes up all the time in several root functions.